File: app/src/main/java/com/gas/app/ui/fragment/mine/mvp/MineModle.java
package com.gas.app.ui.fragment.mine.mvp;

import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.OnLifecycleEvent;

import com.base.lib.di.scope.FragmentScope;
import com.base.lib.integration.repository.IRepositoryManager;
import com.base.lib.mvp.BaseModel;

import javax.inject.Inject;

import timber.log.Timber;

@FragmentScope
public class MineModle extends BaseModel implements MineContract.Model {

    @Inject
    public MineModle(IRepositoryManager repositoryManager) {
        super(repositoryManager);
    }


    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    void onPause() {
        Timber.d("Release Resource");
    }



}

File: app/src/main/java/com/gas/app/ui/fragment/mine/mvp/MineModle.kt
package com.gas.app.ui.fragment.mine.mvp

import androidx.lifecycle.Lifecycle
import androidx.lifecycle.OnLifecycleEvent
import com.base.lib.di.scope.FragmentScope
import com.base.lib.integration.repository.IRepositoryManager
import com.base.lib.mvp.BaseModel
import timber.log.Timber
import javax.inject.Inject

@FragmentScope
class MineModle @Inject constructor(repositoryManager: IRepositoryManager) : BaseModel(repositoryManager), MineContract.Model {
    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    fun onPause() {
        Timber.d("Release Resource")
    }
}